### DISCUSSION TITLE
libmesh installation error for BISON

### DISCUSSION THREAD

**Discussion by aahalimi** [idaholab/moose#26361] [News]
Hello, 

I have been having problems with BISON installation on my HPC session. In particular, compiling libmesh generates the following error:

….
../../include/enums/enum_eigen_solver_type.h ../../include/enums/enum_elem_quality.h ../../include/enums/enum_elem_type.h ../../include/enums/enum_error_estimator_type.h ../../include/enums/enum_fe_family.h ../../include/enums/enum_inf_map_type.h ../../include/enums/enum_io_package.h ../../include/enums/enum_matrix_build_type.h ../../include/enums/enum_norm_type.h ../../include/enums/enum_order.h ../../include/enums/enum_parallel_type.h '/apps/herd/bison/bison-2023-11-30/libmesh/include/libmesh'
cp: cannot create regular file '/apps/herd/bison/bison-2023-11-30/libmesh/include/libmesh/_inst.75162_': Permission denied
make[3]: *** [Makefile:1161: install-includeHEADERS] Error 1
make[3]: Leaving directory '/home/haliassi/workspace/bison/moose/libmesh/build/include'
make[2]: *** [Makefile:1354: install-am] Error 2
make[2]: Leaving directory '/home/haliassi/workspace/bison/moose/libmesh/build/include'
make[1]: *** [Makefile:1188: install-recursive] Error 1
make[1]: Leaving directory '/home/haliassi/workspace/bison/moose/libmesh/build/include'
make: *** [Makefile:33431: install-recursive] Error 1

Can you help me figure out what is the root of such error? 

Thanks, 
Assil

**Comment by milljm** [comment]
Converting this to a discussion, as it is not an issue that can be fixed within the MOOSE repository.

The following path:
```bash
/apps/herd/bison/bison-2023-11-30
```
Indicates you are attempting to use the pre-compiled version of Bison. That said, there is no reason for you to try and build libMesh.

If you are indeed trying to use your own version of Bison, you must refrain from loading the bison module. And instead load the modules that were used to _create_ Bison instead:

```bash
module purge
module load use.moose moose-dev
```

After doing the above, it will be necessary to clean out your previous build, as some pieces of it, will not doubt still be linked to the pre-compiled version. Take care with the following commands. They will delete and non-tracked files you may be working on:

```bash
cd workspace/bison
git clean -xfd
git submodule foreach --recursive git clean -xfd
```

Next, it will be necessary to build PETSc and libMesh, and then Bison:
```bash
cd moose
export MOOSE_JOBS=6 METHODS=opt METHOD=opt
scripts/update_and_rebuild_petsc.sh
scripts/update_and_rebuild_libmesh.sh
scripts/update_and_rebuild_wasp.sh
cd ../
make -j $MOOSE_JOBS
```
